### ISSUE TITLE
TypeError: Cannot create proxy with a non-object as target or handler in 2.6.0

### ISSUE THREAD

**Issue by boldurean** [kettanaito/msw#2342]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Node.js version

20.12.0

### Reproduction repository

https://codesandbox.io/p/devbox/lrh2rd

### Reproduction steps

1. change msw version to 2.4.11 and run `npm test` - test passes
2. change msw version to 2.6.0 and run `npm test` -  test fail

### Current behavior

```JavaScript
TypeError: Cannot create proxy with a non-object as target or handler
 ❯ _WebSocketInterceptor.setup node_modules/.pnpm/@mswjs+interceptors@0.36.7/node_modules/@mswjs/interceptors/src/interceptors/WebSocket/index.ts:70:28
 ❯ _WebSocketInterceptor.apply node_modules/.pnpm/@mswjs+interceptors@0.36.7/node_modules/@mswjs/interceptors/src/Interceptor.ts:130:10
 ❯ SetupServerApi.listen node_modules/.pnpm/msw@2.6.0_@types+node@22.8.5_typescript@5.4.5/node_modules/msw/src/node/SetupServerCommonApi.ts:129:26
 ❯ test.setup.js:6:10
      4| 
      5| beforeAll(() => {
      6|   server.listen({
       |          ^
      7|     onUnhandledRequest(request) {
      8|       console.log("Unhandled %s %s", request.method, request.url);
```

### Expected behavior

All test work pass as before.

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I've modified the `SetupServerCommonApi` class to handle WebSocket interception more gracefully. The changes include:

1. Adding a check to see if WebSocket interception is supported before attempting to set it up.
2. Wrapping the WebSocket interception setup in a try-catch block to prevent errors from breaking the entire setup process.
3. Adding a warning message if WebSocket interception fails to set up.

These changes should prevent the TypeError from occurring while still allowing the rest of the server setup to proceed.

I also created pull request: https://github.com/mswjs/msw/pull/2343


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

**Comment by Afsoon** [comment]
I have reported the same error in the interceptors repository, and I have opened a PR to fix it https://github.com/mswjs/interceptors/pull/669. TL;DR: My findings was, they fixed an error for NextJS, but introduced an error for non NextJS projects.

**Comment by boldurean** [comment]
> I have reported the same error in the interceptors repository, and I have opened a PR to fix it [mswjs/interceptors#669](https://github.com/mswjs/interceptors/pull/669). TL;DR: My findings was, they fixed an error for NextJS, but introduced an error for non NextJS projects.

I would expect simple test will catch it up :|

**Comment by stevensacks** [comment]
Broken for Remix project, as well.

**Comment by custardcream98** [comment]
can't use msw on tests due to this issue.

**Comment by kettanaito** [comment]
I wish we wouldn't have to do these shenanigans. Having a global property descriptor with a getter that returns undefined is a broken implementation on the framework's side and they mustn't do that. I've already opened a pull request to Next.js to fix that (their issue is different), I didn't expect other frameworks to have this. 

Will adjust the interceptor predicate, keeping it sane, hopefully.

**Comment by kettanaito** [comment]
This has been fixed in `@mswjs/interceptors@0.36.8` ([release notes](https://github.com/mswjs/interceptors/releases/tag/v0.36.8)). We can now handle all sorts of property descriptor combinations.

Re-install your dependencies to pull the latest version of this transient dependency to get the fix.
